packages/flutter/lib/src/widgets/fade_in_image.dart

```diff
@@ -463,6 +459,7 @@ class _AnimatedFadeOutFadeIn extends ImplicitlyAnimatedWidget {
     required this.fadeOutCurve,
     required this.fadeInDuration,
     required this.fadeInCurve,
+    required this.wasSynchronouslyLoaded,
   }) : assert(target != null),
```

Comment:
maybe add an assert if wasSynchronouslyLoaded is true, isTargetLoaded must also be true

Reply:
assert(! wasSynchronouslyLoaded || isTargetLoaded)

Reply:
done